kubectl-volsync/cmd/migration_create.go

```diff
@@ -72,34 +108,294 @@ func validateMigrationCreate(cmd *cobra.Command, args []string) error {
+	// create destination PVC if does not exist
+	if mc.PVC == nil {
+		mc.PVC, err = mc.createDestinationPVC(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	// create migration destination
+	err = mc.createDestination(ctx)
 	if err != nil {
 		return err
 	}
+	// save the destination details into config file
+	if err = mc.mr.Save(); err != nil {
+		return fmt.Errorf("unable to save relationship configuration: %w", err)
+	}
+	return nil
+}
 
-	// Insert information into the relationship & save it
-	cap, err := cmd.Flags().GetString("capacity")
-	if err != nil {
+func (mc *migrationCreate) createNamespace(ctx context.Context) (*v1.Namespace, error) {
+	mrd := mc.mr.data.Destination
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mrd.Namespace,
+		},
+	}
+	if err := mc.clientObject.Create(ctx, ns); err != nil {
+		if kerrs.IsAlreadyExists(err) {
+			klog.Infof("Namespace: \"%s\" already present, proceeding with this namespace",
+				mrd.Namespace)
+			return ns, nil
+		}
+		return nil, err
+	}
+	klog.Infof("Created destination namespace: \"%s\"", mrd.Namespace)
+
+	return ns, nil
+}
+
+func (mc *migrationCreate) createDestinationPVC(ctx context.Context) (*v1.PersistentVolumeClaim, error) {
+	mrd := mc.mr.data.Destination
+	var destPVC *v1.PersistentVolumeClaim
+
+	if *mrd.Destination.StorageClassName == "" {
+		destPVC = &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      mrd.PVCName,
+				Namespace: mrd.Namespace,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				AccessModes: mrd.Destination.AccessModes,
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: *mrd.Destination.Capacity,
+					},
+				},
+			},
+		}
+	} else {
+		destPVC = &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      mrd.PVCName,
+				Namespace: mrd.Namespace,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				AccessModes:      mrd.Destination.AccessModes,
+				StorageClassName: mrd.Destination.StorageClassName,
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: *mrd.Destination.Capacity,
+					},
+				},
+			},
+		}
+	}
+	if err := mc.clientObject.Create(ctx, destPVC); err != nil {
+		return nil, err
+	}
+	mrd.Destination.StorageClassName = destPVC.Spec.StorageClassName
+	klog.Infof("Created destination PVC: \"%s\"", mrd.PVCName)
+
+	return destPVC, nil
+}
+
+func (mc *migrationCreate) getDestinationPVC(ctx context.Context) *v1.PersistentVolumeClaim {
+	mrd := mc.mr.data.Destination
+	destPVC := &v1.PersistentVolumeClaim{}
+	pvcInfo := types.NamespacedName{
+		Namespace: mrd.Namespace,
+		Name:      mrd.PVCName,
+	}
+	err := mc.clientObject.Get(ctx, pvcInfo, destPVC)
+	if err == nil {
+		return destPVC
+	}
+
+	return nil
+}
+
+func (mc *migrationCreate) createDestination(ctx context.Context) error {
+	mrd := mc.mr.data.Destination
+	rsyncSpec := &volsyncv1alpha1.ReplicationDestinationRsyncSpec{
```

Comment:
My expectation had been that `data.Destination` was just going to get directly assigned to `rsyncSpec` as opposed to only some of the fields. I think a straight assignment would probably make it a bit more obvious elsewhere in the code (e.g. newMigrationRelationshipDestination) how the values were going to be used.

Reply:
sure. I will assign rsyncSpec coming from top and not build it here